src/polly/src/event_manager.rs

```diff
@@ -289,14 +287,21 @@ impl EventManager {
     /// Register a new event handler.
     /// handler.init() will specify the pollable and event set.
     ///
+    /// Returns the handler wrapped in an Arc<Mutex>.
     pub fn register<T: EventHandler + 'static>(&mut self, handler: T) -> Result<Arc<Mutex<T>>> {
-        let pollable_ops = handler.init();
-        let wrapped_type = Arc::new(Mutex::new(handler));
-        let wrapped_handler: Arc<Mutex<dyn EventHandler>> = wrapped_type.clone();
-
-        self.update(wrapped_handler, pollable_ops)?;
+        let wrapped_handler = Arc::new(Mutex::new(handler));
+        self.register_protected(wrapped_handler.clone())?;
+        Ok(wrapped_handler)
+    }
 
-        Ok(wrapped_type)
+    /// Register an event handler object already wrapped in an Arc<Mutex>.
+    pub fn register_protected<T: EventHandler + 'static>(
```

Comment:
I think another option here would be to only have the register() function which accepts Arc<Mutex<T>> and create a static function that receives T and returns Arc<Mutex<T>>.

Reply:
I'm not sure I follow. What would be the signatures of these functions? What would be better/clearer in that case?

Reply:
I think this would clear up things:
```rust
pub fn wrap_handler<T: EventHandler + 'static>(handler: T) -> Arc<Mutex<T>>;
pub fn register<T: EventHandler + 'static>(mut &self, handler: Arc<Mutex<T>>) -> Result<()>;

// You can use it like this:
em.register(EventManager::wrap_handler(handler));
// or in your case:
let handler = EventManager::wrap_handler(serial_device);
em.register(EventManager::wrap_handler(handler.clone());

```

Reply:
Sure, that should work.

Reply:
Done, check it out.